File: pyg-sellergoods-interface/src/main/java/com/pyg/sellergoods/service/GoodsService.java
package com.pyg.sellergoods.service;

import PageBean.PageResult;
import com.pyg.pojo.TbGoods;
import com.pyg.pojo.TbItem;
import com.pyg.pojogroup.Goods;

import java.util.List;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface GoodsService {

    /**
     * 返回全部列表
     *
     * @return
     */
    public List<TbGoods> findAll();


    //更新数据
    void update(Goods goods);

    //    新增数据
    void add(Goods goods);


    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    public Goods findOne(Long id);


    /**
     * 批量删除
     *
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 分页
     *
     * @param pageNum  当前页 码
     * @param pageSize 每页记录数
     * @return
     */
    PageResult search(TbGoods goods, int pageNum, int pageSize);

    //审核状态更新
    public void updateStatus(Long[] ids, String status);

    /**
     * 根据商品 ID 和状态查询 Item 表信息
     * @param goodsIds
     * @param status
     * @return
     */
    public List<TbItem> findItemListByGoodsIdandStatus(Long[] goodsIds, String status );

}

File: pyg-manager-web/src/main/webapp/js/Service/goodsService.js
//服务层
app.service('goodsService', function ($http) {

    //读取列表数据绑定到表单中
    this.findAll = function () {
        return $http.get('../goods/findAll');
    }
    //存储&修改
    this.save = function (entity) {
        return $http.post('../goods/save', entity);
    }

    //查询实体
    this.findOne = function (id) {
        return $http.get('../goods/findOne?id=' + id);
    }

    //删除
    this.dele = function (ids) {
        return $http.get('../goods/delete?ids=' + ids);
    }


    //分页 查询
    this.findPage = function (page, size, searchEntity) {
        return $http.post('../goods/search?page=' + page + '&size=' + size, searchEntity);
    }

    //更新状态
    this.updateStatus=function (ids,status) {
        return $http.get('../goods/updateStatus?ids='+ids+"&status="+status);

    }
});

